renderlib/graphics/RenderGL.cpp

```diff
@@ -100,11 +103,60 @@ RenderGL::renderTo(const CCamera& camera, GLFramebufferObject* fbo)
 
   doClear();
   if (haveScene) {
-    m_image3d->render(camera, m_scene, m_renderSettings);
+    drawSceneObjects(camera);
   }
   fbo->release();
 }
 
+void
+RenderGL::drawSceneObjects(const CCamera& camera)
+{
+  const glm::vec3 volumePhysicalSize = m_scene->m_volume->getPhysicalDimensions();
+  float maxPhysicalDim = std::max(volumePhysicalSize.x, std::max(volumePhysicalSize.y, volumePhysicalSize.z));
+
+  // scene bounds are min=0.0, max=image physical dims scaled to max dim so that max dim is 1.0
+  glm::vec3 sn = m_scene->m_boundingBox.GetMinP();
+  glm::vec3 ext = m_scene->m_boundingBox.GetExtent();
+  CBoundingBox b;
+  b.SetMinP(glm::vec3(ext.x * m_scene->m_roi.GetMinP().x + sn.x,
+                      ext.y * m_scene->m_roi.GetMinP().y + sn.y,
+                      ext.z * m_scene->m_roi.GetMinP().z + sn.z));
+  b.SetMaxP(glm::vec3(ext.x * m_scene->m_roi.GetMaxP().x + sn.x,
+                      ext.y * m_scene->m_roi.GetMaxP().y + sn.y,
+                      ext.z * m_scene->m_roi.GetMaxP().z + sn.z));
+  // LOG_DEBUG << "CLIPPED BOUNDS" << b.ToString();
+  // LOG_DEBUG << "FULL BOUNDS" << m_scene->m_boundingBox.ToString();
+  // draw bounding box on top.
+  // move the box to match where the camera is pointed
+  // transform the box from -1..1 to 0..physicalsize
+  float maxd = (std::max)(ext.x, (std::max)(ext.y, ext.z));
+  glm::vec3 scales(0.5 * ext.x / maxd, 0.5 * ext.y / maxd, 0.5 * ext.z / maxd);
+  // it helps to imagine these transforming the space in reverse order
+  // (first translate by 1.0, and then scale down)
+  glm::mat4 bboxModelMatrix = glm::scale(glm::mat4(1.0f), scales);
+  bboxModelMatrix = glm::translate(bboxModelMatrix, glm::vec3(1.0, 1.0, 1.0));
+  glm::mat4 viewMatrix(1.0);
+  glm::mat4 projMatrix(1.0);
+  camera.getProjMatrix(projMatrix);
+  camera.getViewMatrix(viewMatrix);
+
+  if (m_scene->m_material.m_showBoundingBox) {
```

Comment:
I'm seeing some other flags get enabled in RenderGLPT.cpp that aren't here (`GL_DEPTH_TEST`, Depth Mask, Color Mask, etc.). Are those not required here because we can directly draw the lines in the scene, and we don't have to worry about sorting with the pathtraced volumes?

Reply:
Yes, those settings are differences in how the pathtrace renderer draws itself.  Raymarch is simpler because it's drawing a box geometry, whereas pathtrace is basically drawing a flat fullscreen buffer that needs to be composited correctly as if it is inside the bounding box lines.